### ISSUE TITLE
[Bug]: `addon-a11y-addon-test` migration fails if using csf factories

### ISSUE THREAD

**Issue by mrginglymus** [storybookjs/storybook#31940]
### Describe the bug

Running the above migration against a storybook using csf factories fails:

```
  1) We couldn't find or automatically update .storybook/vitest.setup.<ts|js> in your project to smoothly set up     │
│   project annotations from @storybook/addon-a11y.                                                                    │
│   Please manually update your vitest.setup.ts file to include the following:                                         │
│                                                                                                                      │
│   ...                                                                                                                │
│   + import * as a11yAddonAnnotations from "@storybook/addon-a11y/preview";                                           │
│                                                                                                                      │
│   setProjectAnnotations([                                                                                            │
│   ...                                                                                                                │
│   + a11yAddonAnnotations,                                                                                            │
│   ]);                      
```
unsurprisingly, as the above snippet is irrelevant to csf factories

### Reproduction link

-

### System

```bash
-
```
